lib/Dialect/SV/SVOps.cpp

```diff
@@ -1082,6 +1083,94 @@ void ArrayIndexInOutOp::build(OpBuilder &builder, OperationState &result,
   build(builder, result, InOutType::get(resultType), input, index);
 }
 
+//===----------------------------------------------------------------------===//
+// IndexedPartSelectInOutOp
+//===----------------------------------------------------------------------===//
+
+void IndexedPartSelectInOutOp::build(OpBuilder &builder, OperationState &result,
+                                     Value input, Value base, int32_t width,
+                                     bool decrement) {
+  auto resultType =
+      hw::InOutType::get(IntegerType::get(builder.getContext(), width));
+  build(builder, result, resultType, input, base, width, decrement);
+}
+
+LogicalResult IndexedPartSelectInOutOp::inferReturnTypes(
+    MLIRContext *context, Optional<Location> loc, ValueRange operands,
+    DictionaryAttr attrs, mlir::RegionRange regions,
+    SmallVectorImpl<Type> &results) {
+  auto width = attrs.get("width");
+  if (!width)
+    return failure();
+
+  results.push_back(hw::InOutType::get(
+      IntegerType::get(context, width.cast<IntegerAttr>().getInt())));
+  return success();
+}
+
+//===----------------------------------------------------------------------===//
+// IndexedPartSelectOp
+//===----------------------------------------------------------------------===//
+
+void IndexedPartSelectOp::build(OpBuilder &builder, OperationState &result,
+                                Value input, Value base, int32_t width,
+                                bool decrement) {
+  auto resultType = (IntegerType::get(builder.getContext(), width));
+  build(builder, result, resultType, input, base, width, decrement);
+}
+
+LogicalResult IndexedPartSelectOp::inferReturnTypes(
+    MLIRContext *context, Optional<Location> loc, ValueRange operands,
+    DictionaryAttr attrs, mlir::RegionRange regions,
+    SmallVectorImpl<Type> &results) {
+  auto width = attrs.get("width");
+  if (!width)
+    return failure();
+
+  results.push_back(
+      IntegerType::get(context, width.cast<IntegerAttr>().getInt()));
+  return success();
+}
+
+static LogicalResult verifyIndexedPartSelectOp(Operation *op) {
+  return TypeSwitch<Operation *, LogicalResult>(op)
+      .Case<IndexedPartSelectOp, IndexedPartSelectInOutOp>(
+          [&](auto p) -> LogicalResult {
+            unsigned inputWidth = 0, resultWidth = 0;
+            auto width = p.width();
+            if (isa<IndexedPartSelectInOutOp>(p)) {
+              if (auto i = p.input()
+                               .getType()
+                               .template cast<InOutType>()
+                               .getElementType()
+                               .template dyn_cast<IntegerType>())
+                inputWidth = i.getWidth();
+              else
+                return op->emitError("input element type must be Integer");
+              if (auto resType = p.getType()
```

Comment:
Are tehse emitErrors's reachable?  The verify hook for an op is only invoked if all the ODS predicates pass, and they already constrain the types to integers.